infra/modules/aws/main.tf

```diff
@@ -0,0 +1,107 @@
+# setup AWS project to host Psoxy instances
+
+terraform {
+  required_providers {
+    aws = {
+      version = "~> 3.0"
+    }
+  }
+}
+
+# AWS API Gateway
+resource "aws_apigatewayv2_api" "psoxy-api" {
+  name          = "psoxy-api"
+  protocol_type = "HTTP"
+  description   = "API to expose psoxy instances"
+}
+
+resource "aws_cloudwatch_log_group" "gateway-log" {
+  name = aws_apigatewayv2_api.psoxy-api.name
+}
+
+resource "aws_apigatewayv2_stage" "live" {
+  api_id        = aws_apigatewayv2_api.psoxy-api.id
+  name          = "live" # q: what name??
+  auto_deploy   = true
+  access_log_settings {
+    destination_arn = aws_cloudwatch_log_group.gateway-log.arn
+    format          = "$context.identity.sourceIp $context.identity.caller $context.identity.user [$context.requestTime] \"$context.httpMethod $context.resourcePath $context.protocol\" $context.status $context.responseLength $context.requestId $context.extendedRequestId"
+  }
+}
+
+# role that Worklytics user will use to call the API
+resource "aws_iam_role" "api-caller" {
+  name = "PsoxyApiCaller"
+
+  # who can assume this role
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" = "sts:AssumeRole"
+        "Effect" : "Allow"
+        "Principal" : {
+          "AWS" : "arn:aws:iam::${var.caller_aws_account_id}"
+        }
+        #TODO: add condition referencing GCP service account that will auth with Worklytics AWS account to call
```

Comment:
watch out, if adding this the test call won't work. We'd need to add another policy for the GCP case.